static/app/views/releases/thresholdsList/thresholdGroupRows.tsx

```diff
@@ -213,56 +206,45 @@ export function ThresholdGroupRows({
   return (
     <StyledThresholdGroup>
       {Array.from(thresholdIdSet).map((tId: string, idx: number) => {
-        const threshold = editingThresholds[tId] || thresholdsById[tId];
-        let environmentName;
-        if (editingThresholds[tId]) {
-          // editing environment type is String
-          environmentName = editingThresholds[tId].environment;
+        let threshold;
+        let isEditing = false;
+        if (tId in editingThresholds) {
+          isEditing = true;
+          threshold = editingThresholds[tId] as EditingThreshold;
         } else {
-          // Threshold environment type is Environnment
-          environmentName = thresholdsById[tId].environment
-            ? thresholdsById[tId].environment.name
-            : '';
+          threshold = initialThreshold as Threshold;
         }
+
         return (
           <StyledRow
             key={threshold.id}
-            lastRow={idx === thresholdIdSet.size - 1}
-            hasError={threshold.hasError}
+            lastRow={isLastRow && idx === thresholdIdSet.size - 1}
+            hasError={isEditing && (threshold as EditingThreshold).hasError}
           >
-            <FlexCenter style={{borderBottom: 0}}>
-              {idx === 0 ? (
-                <ProjectBadge
-                  project={threshold.project}
-                  avatarSize={16}
-                  hideOverflow
-                  disableLink
-                />
-              ) : (
-                ''
-              )}
-            </FlexCenter>
-            {/* TODO: IF this is a new threshold - allow environment select */}
-            {newGroup ? (
+            {/* ENV ONLY EDITABLE IF NEW */}
+            {!initialThreshold || threshold.id !== initialThreshold.id ? (
               <CompactSelect
                 style={{width: '100%'}}
-                value={threshold.environment}
+                value={threshold.environmentName}
                 onChange={selectedOption =>
                   editThresholdState(threshold.id, 'environment', selectedOption.value)
                 }
-                options={newGroup.environments.map(env => ({
+                options={allEnvironmentNames.map(env => ({
                   value: env,
                   textValue: env,
                   label: env,
                 }))}
               />
             ) : (
-              <FlexCenter style={{borderBottom: 0}}>
-                {idx === 0 ? environmentName || 'None' : ''}
+              <FlexCenter>
+                {/* 'None' means it _has_ an environment, but the env has no name */}
+                {threshold.environment
+                  ? threshold.environment.name || 'None'
+                  : '{No environment}'}
```

Comment:
if the threshold has no environment does it apply to all environments?

Reply:
Nope - because you can have releases that do not have an environment to them as well

Reply:
Does this actually happen? If so can you add the check on threshold creation to make sure it can have no env but it can't have an env with no name?

Reply:
This is an edge case i've called out in the BE side as well.

currently we don't differentiate between an environment with no Name, vs. a Release with no environment.
So - creating a threshold with None set to environment will actually apply to both of those cases. For now i think this is fine